### PR TITLE
Adding functionality to Prometheus to have hipchat alerts

### PR DESCRIPTION
**Pull request by rekhajoshm**
Adding functionality to Prometheus to have hipchat alerts.Thanks.

### CODE REVIEW

**Comment by rekhajoshm** [comment]
correcting local git user

File: config/config.proto [hunk 1]
```diff
@@ -34,6 +34,16 @@ message PushoverConfig {
   optional string user_key = 2;
 }
 
+// Configuration for notification via Hipchat.
+message HipChatConfig {
+  // Hipchat service key, see:
+  //https://www.hipchat.com
```
Comment (brian-brazil) [inline config/config.proto]: space after the //

File: manager/notifier.go [hunk 2]
```diff
@@ -276,6 +278,46 @@ func (n *notifier) sendPushoverNotification(token, userKey string, a *Alert) err
 	return err
 }
 
+func (n *notifier) sendHipChatNotification(token, room string, a *Alert) error {
+	// https://www.hipchat.com/
+        incidentKey := a.Fingerprint()
+        buf, err := json.Marshal(map[string]interface{}{
```
Comment (brian-brazil) [inline manager/notifier.go]: Could we make this more human readable? I'd go for description and labels.

File: config/fixtures/duplicate_nc_name.conf.input [hunk 3]
```diff
@@ -6,6 +6,10 @@ notification_config {
   email_config {
     email: "test@testservice.org"
   }
+  hipchat_config {
```
Comment (brian-brazil) [inline config/fixtures/duplicate_nc_name.conf.input]: Are you using this in any tests?

File: README.md [hunk 4]
```diff
@@ -11,7 +11,8 @@ following aspects:
 * handling notification repeats
 * sending alert notifications via external services (currently email,
 [PagerDuty](http://www.pagerduty.com/) or
-[Pushover](https://www.pushover.net/))
+[Pushover](https://www.pushover.net/) or 
+[HipChat] (https://www.hipchat.com/))
```
Comment (juliusv) [inline README.md]: No space between `](` please.

File: config/config.proto [hunk 5]
```diff
@@ -44,6 +54,9 @@ message NotificationConfig {
   repeated EmailConfig email_config = 3;
   // Zero or more pushover notification configurations.
   repeated PushoverConfig pushover_config = 4;
+  // Zero or more HipChat notification configurations.
+  repeated HipChatConfig hipchat_config = 5;
+
```
Comment (juliusv) [inline config/config.proto]: remove the extra empty line please

File: manager/notifier.go [hunk 6]
```diff
@@ -56,6 +57,7 @@ var (
 	pagerdutyApiUrl        = flag.String("pagerdutyApiUrl", "https://events.pagerduty.com/generic/2010-04-15/create_event.json", "PagerDuty API URL.")
 	smtpSmartHost          = flag.String("smtpSmartHost", "", "Address of the smarthost to send all email notifications to.")
 	smtpSender             = flag.String("smtpSender", "alertmanager@example.org", "Sender email address to use in email notifications.")
+	hipchatApiUrl	       = flag.String("hipchatApiUrl","https://api.hipchat.com/v1/", "Hipchat API URL");
```
Comment (juliusv) [inline manager/notifier.go]: This flag name is good for now, but I just noticed that this file is still using an old style of flag names and doesn't follow the convention we use everywhere else in Prometheus now (lower-case names with dots as namespace separators). So expect a flag name change here soon...

File: manager/notifier.go [hunk 7]
```diff
@@ -276,6 +278,46 @@ func (n *notifier) sendPushoverNotification(token, userKey string, a *Alert) err
 	return err
 }
 
+func (n *notifier) sendHipChatNotification(token, room string, a *Alert) error {
+	// https://www.hipchat.com/
```
Comment (juliusv) [inline manager/notifier.go]: How about `// See https://www.hipchat.com/docs/api.`?

File: manager/notifier.go (hunk 2 above)
Comment (juliusv) [inline manager/notifier.go]: @brian-brazil Not sure what you mean here? It's basically just doing the same as the code for PagerDuty?

Comment (juliusv) [inline manager/notifier.go]: Hmm, now I get it. I thought this was constructing JSON for the API to consume. But it just becomes a message to be read by humans (incidentally in the PagerDuty JSON API format). Indeed, it should probably have a more human-readable content then.

**Comment by juliusv** [comment]
Heya, thanks for sending this! Compiling this change still fails with a bunch of compile errrors:

```
.deps/gopath/src/github.com/prometheus/alertmanager/manager/notifier.go:298: invalid indirect of token (type string)
.deps/gopath/src/github.com/prometheus/alertmanager/manager/notifier.go:298: invalid indirect of room (type string)
.deps/gopath/src/github.com/prometheus/alertmanager/manager/notifier.go:300: not enough arguments to return
.deps/gopath/src/github.com/prometheus/alertmanager/manager/notifier.go:302: invalid indirect of token (type string)
.deps/gopath/src/github.com/prometheus/alertmanager/manager/notifier.go:304: cannot use bytes.NewBuffer(buf) (type *bytes.Buffer) as type string in field value
.deps/gopath/src/github.com/prometheus/alertmanager/manager/notifier.go:305: invalid indirect of room (type string)
```

Could you fix those first?

**Comment by juliusv** [comment]
Closing in favor of https://github.com/prometheus/alertmanager/pull/45.

**Pull request closed without merging**